lib/ui/tabs/shared/outgoing_album_item.dart

```diff
@@ -0,0 +1,116 @@
+import "package:flutter/material.dart";
+import "package:photos/generated/l10n.dart";
+import "package:photos/models/collection_items.dart";
+import "package:photos/models/gallery_type.dart";
+import 'package:photos/theme/colors.dart';
+import "package:photos/ui/viewer/file/thumbnail_widget.dart";
+import "package:photos/ui/viewer/gallery/collection_page.dart";
+import "package:photos/utils/navigation_util.dart";
+
+class OutgoingAlbumItem extends StatelessWidget {
+  final CollectionWithThumbnail c;
+
+  const OutgoingAlbumItem({super.key, required this.c});
+
+  @override
+  Widget build(BuildContext context) {
+    final shareesName = <String>[];
+    if (c.collection.hasSharees) {
+      for (int index = 0; index < c.collection.sharees!.length; index++) {
+        final sharee = c.collection.sharees![index]!;
+        final String name =
+            (sharee.name?.isNotEmpty ?? false) ? sharee.name! : sharee.email;
+        if (index < 2) {
+          shareesName.add(name);
+        } else {
+          final remaining = c.collection.sharees!.length - index;
+          if (remaining == 1) {
+            // If it's the last sharee
+            shareesName.add(name);
+          } else {
+            shareesName.add(
+              "and " +
+                  remaining.toString() +
+                  " other" +
+                  (remaining > 1 ? "s" : ""),
+            );
+          }
+          break;
+        }
+      }
+    }
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      child: Container(
+        margin: const EdgeInsets.fromLTRB(16, 12, 16, 12),
+        child: Row(
+          children: <Widget>[
+            ClipRRect(
+              borderRadius: BorderRadius.circular(1),
+              child: SizedBox(
+                height: 60,
+                width: 60,
+                child: Hero(
+                  tag: "outgoing_collection" + c.thumbnail!.tag,
+                  child: ThumbnailWidget(
+                    c.thumbnail,
+                    key: Key("outgoing_collection" + c.thumbnail!.tag),
```

Comment:
Can use ValueKey()

Reply:
Planning to modify these later. The diff only has rename refactoring, no logic change (i think)